### ISSUE TITLE
Make edgemarks easier to create

### ISSUE THREAD

**Issue by gereleth** [gereleth/hexapipes#71]
I can't always create an edgemark on the first try... Sometimes not even on the second one.

I've tried relaxing the conditions around placing an edgemark so that almost any small drag goes. This is both for mouse drags and touch drags.

A preview link to try it out: https://hexapipes-git-controls-fixes-gereleth.vercel.app/

Interested in feedback: are the edgemarks easier to create there? Do you make them by accident? Any marks appearing in the wrong place? And overall - is this an improvement over the current situation? =)

**Comment by gordonwoodhull** [comment]
I did find the feature easier to use. However I also ended up with a lot of unintentional marks. 🤷‍♂️ 

TBH I don’t use this feature all that much. I will “prove” enough of the board and lock it to get started, and then guess the rest. My times are not impressive but even so I don’t find time for half-measures.

**Comment by gereleth** [comment]
I don't use it a lot either any more but I sure did before I internalized enough patterns to get by on just guessing! I think it's very useful when starting out and making your first deductions.

I also use edgemarks in tricky puzzles as a part of poor girl's "trial mode". I use locked tiles as "suppose this goes here then this goes here". I unlock them all when a trial goes wrong. And I store things I definitely know in the edgemarks =).

I increased the drag distance required for an edgemark a little, will do some more testing.

**Comment by Jaybrainer** [comment]
alternative is to add a toggle which enables "edge marks only" mode, where left/right-click, or a cyclic input for mobile would place edge marks with a simple click. This can be a hotkey on PC, but needs a toggle for mobile, which should be easily accessible in the eventual UI overhaul.

**Comment by gereleth** [comment]
While experimenting with edgemarks I made this "fingerpainting mode" - you press and hold <kbd>X</kbd> key and then click and drag to show where the pipes should go. This creates connection marks and tiles rotate to match them.

https://user-images.githubusercontent.com/6080076/219896394-72715dc2-3891-43fc-b413-d4afa6b312ea.mp4

This was a fun experiment but I didn't find this mode intuitive enough to switch. Making guesses is tricky, so is fixing mistakes. I also don't know how to bring this to touch devices so I'm not pursuing it further.

But then I thought that when you lock a tile it's essentially the same as setting all its edgemarks. And maybe the neighbours should all rotate to match in this case too. This inspired the smart assistant feature:

https://user-images.githubusercontent.com/6080076/219897744-0df255d8-2044-4345-bfe2-1745097388ae.mp4

I played with the assistant a lot in the preview branch and I sorely miss it on the main site already)). It feels like a chaos-banishing superpower! My best 5x5 time is down to 6 seconds because some lucky puzzles can be solved in just a single locking sweep.
I'm planning to merge it in soon)).
It's off by default but can be enabled in the settings.
